Sources/AppcuesKit/Presentation/UI/AppcuesView.swift

```diff
@@ -0,0 +1,121 @@
+
+        isConfigured = true
+
+        trackEmbed()
+    }
+
+    private func trackEmbed() {
+        // only track if we have not already, the required props have been configured, and we have a valid embed ID
+        guard !hasTracked, isConfigured, isAttachedToView, let embedId = embedId, !embedId.isEmpty, let appcues = appcues else { return }
+        // this screen call notifies the server that the embed is available for rendering content
+        // re-purposing screen for this in the short term, a deeper topic for how to best handle for embeds
+        appcues.screen(title: embedId)
+    }
+
+    override public func didMoveToSuperview() {
+        super.didMoveToSuperview()
+
+        if subviews.isEmpty {
+            // possibly have a delegate for these embeds that would allow the host app to have more control over this?
+            isHidden = true
+        }
+
+        isAttachedToView = true
+
+        trackEmbed()
+    }
+
+    internal func embed(_ experienceController: UIViewController, margins: NSDirectionalEdgeInsets, animated: Bool) {
+        guard let viewController = viewController else { return }
+
+        heightObserver = experienceController.observe(\.preferredContentSize) { _, _ in
+            self.preferredHeightConstraint.constant = experienceController.preferredContentSize.height
+        }
+
+        self.experienceController = experienceController
+        viewController.embedChildViewController(experienceController, inSuperview: self, margins: margins)
+
+        if animated {
+            UIView.animate(withDuration: 0.3) {
+                // possibly have a delegate for these embeds that would allow the host app to have more control over this?
+                self.isHidden = false
+            }
+        } else {
+            isHidden = false
+        }
+    }
+
+    internal func unembed(animated: Bool) {
+        guard let experienceController = experienceController else { return }
+
+        preferredHeightConstraint.constant = 0
+        // preferredContentSize = .zero
+        heightObserver = nil
+
+        if animated {
+            UIView.animate(
+                withDuration: 0.3,
+                animations: {
+                    // possibly have a delegate for these embeds that would allow the host app to have more control over this?
+                    self.isHidden = true
+                },
+                completion: { _ in
+                    self.viewController?.unembedChildViewController(experienceController)
+                    self.experienceController = nil
+                })
+        } else {
+            isHidden = true
+            viewController?.unembedChildViewController(experienceController)
+            self.experienceController = nil
+        }
+    }
+
+    internal func remove() {
+        guard let viewController = viewController else { return }
+        embedTrait?.remove(viewController: viewController, completion: nil)
+    }
```

Comment:
This is very clever to have skippable call this which calls the embed trait dismisser!